finetune/lm_sequence_labeling.py

```diff
@@ -0,0 +1,121 @@
+
+        return tokens, mask, padded_labels
+
+    def _text_to_ids_with_token_positions(self, *Xs):
+        tokens_ids, tok_pos = self.encoder.encode_sequence_labeling_inferrence(*Xs, max_length=self.hparams.max_length)
+        x, m = self._array_format(tokens_ids)
+
+        return x, m, tok_pos
+
+    def _finetune(self, *Xs, Y, batch_size=None):
+        """
+        X: List / array of text
+        Y: Class labels
+        val_size: Float fraction or int number that represents the size of the validation set.
+        val_interval: The interval for which validation is performed, measured in number of steps.
+        """
+        train_x, train_mask, sequence_labels = self._text_to_ids_with_labels(*Xs)
+        return self._training_loop(train_x, train_mask, sequence_labels, batch_size=batch_size or self.hparams.batch_size)
+
+    def get_target_encoder(self):
+        return SequenceLabelingEncoder()
+
+    def predict_ops(self, logits):
+        return sequence_predict(logits, self.predict_params)
+
+    def _text_to_ids(self, *Xs, max_length=None):
+        """
+        For sequence labeling this is a NOOP. See LanguageModelSequence._text_to_ids* for specific train and predict
+        implementations.
+        """
+        return Xs
+
+    def finetune(self, XYs, batch_size=None):
+        """
+        :param XYs: An array of labeled text snippets. Format: [batch_size, sequences_per_data, snippets_per_sequence, 2]
+            where the final dimension is of the format [text_snippet, label]
+        :param batch_size: integer number of examples per batch. When N_GPUS > 1, this number
+                           corresponds to the number of training examples provided to each GPU.
+        :param val_size: Float fraction or int number that represents the size of the validation set.
+        :param val_interval: The interval for which validation is performed, measured in number of steps.
+        """
+        self.target_type = SEQUENCE_LABELING
+        return self._finetune(*list(zip(*XYs)), Y=None, batch_size=batch_size)
+
+    def predict(self, Xs, max_length=None):
+        """
+        Produces a list of most likely class labels as determined by the fine-tuned model.
+
+        :param Xs: An iterable of lists or array of text, shape [batch, n_inputs, tokens]
+        :param max_length: the number of tokens to be included in the document representation.
+                           Providing more than `max_length` tokens as input will result in truncation.
+        :returns: list of class labels.
+        """
+        sequence_major = list(zip(*Xs))
+        x_pred, m_pred, tok_pos = self._text_to_ids_with_token_positions(*sequence_major)
+        sparse_predictions = self._predict(x_pred, m_pred, max_length=max_length)
+        output = []
+
+        for texts, labels, token_position in zip(Xs, sparse_predictions, tok_pos):
```

Comment:
We can eventually remove this translation to dense predictions -- we have visualization tools for the sparse predictions in Indico API format.